### ISSUE TITLE
Proposed rule: `global` or `nonlocal` looks skippable but applies to the whole function

### ISSUE THREAD

**Issue by andersk** [astral-sh/ruff#6470]
In this example, the `global` declaration is misleading, because Python actually applies it to the whole function, not just the `if` branch it’s inside.

```python
g = 1

def f(b: bool) -> None:
    if b:
        global g
        g = 2
    else:
        g = 3

f(False)
print(g)  # 3, not 1
```

The specific rule I would propose is as follows:

- If there’s a `global g` or `nonlocal g` declaration at location X, and a reference to `g` at location Y, and there exists a syntactic control flow path from the function start to Y that doesn’t go through X, then raise a diagnostic “`global g` looks skippable but applies to the whole function”.

(This formulation avoids unnecessarily flagging the above example if there were no `else: g = 3` branch.)
